vms/platformvm/vm.go

```diff
@@ -193,6 +193,9 @@ func (vm *VM) Initialize(
 		mempool,
 		txExecutorBackend.Config.PartialSyncPrimaryNetwork,
 		appSender,
+		&chainCtx.Lock,
```

Comment:
I should probably have asked [this](https://github.com/ava-labs/avalanchego/pull/3428#discussion_r1831468981) here. 

Should we pass an `RLock` here? Or alternatively pass full RWMutex and do a readlock in warp verifier?

Reply:
It wouldn't be incorrect to pass in `chainCtx.Lock.RLocker()`. But we are pretty liberal with grabbing the write lock in the P-chain, so I felt like this followed the rest of the code...

Looking into the actual state implementation of the methods that are called:
- `GetSubnetConversion`
- `GetSubnetOnlyValidator`
- `GetTimestamp`
- `HasExpiry`

They all seem to be safe to be called with the read lock... We haven't really done a pass to ensure that it's safe to use Rlocks for the getters of the state

Reply:
I think we should probably start using read locks and do a pass to ensure we're not overusing write locks. Given that we started to use them in network/p-chain APIs etc.

Reply:
Do you think it makes sense to do this in this PR? or as a followup?

Reply:
I think we can change the lock here to readlock and then we can do a followup to clean rest of them

Reply:
I changed the locker to be a RLock and documented the functions as being safe for concurrent reads.